pkg/etcd/etcd.go

```diff
@@ -458,27 +458,33 @@ func (c *CDCEtcdClientImpl) CreateChangefeedInfo(ctx context.Context,
 	}
 	opsThen := []clientv3.Op{
 		clientv3.OpPut(infoKey, value),
-		clientv3.OpPut(upstreamEtcdKeyStr, string(upstreamData)),
 	}
 	if len(upstreamResp.Kvs) == 0 {
 		cmps = append(cmps,
 			clientv3.Compare(clientv3.ModRevision(upstreamEtcdKeyStr),
 				"=", 0))
+		opsThen = append(opsThen,
+			clientv3.OpPut(upstreamEtcdKeyStr, string(upstreamData)))
```

Comment:
Should we consider the case that this function(`CreateChangefeedInfo`) was called concurrently, is it still possible for it to start as `upstreamResp.Kvs` is empty and then become non-empty?
for example:
goroutine1 at t1: upstreamResp.kvs is empty 
goroutine2 at t2:upstreamResp.kv is empty
goroutine1 comes to L466 and finished L478 successfully
goroutine2 comes to L466 and append this PutOp and goes to L478 failed.
Meanwhile, could you please add a failpoint for it and add a unit test for the above case?

Reply:
The multi-upstream feature has not been released, so this issue does not exist in customer scenarios. 

Note that there are two apis that update or create upstream information:
1. create changfeed and upstream`PUT /api/v2/changefeeds/{changefeed_id}`
2. update changefeed and upstream `POST /api/v2/changefeeds`

And, the expected usage of these apis is:
1. Avoid modifying the same changefeed concurrently (This is obvious).
2. Avoid modifying the same upstream concurrently (However, this is not obvious).

The second scenario here is described in the comments. 

In the future, perhaps we should update the upstream information using a separate api.